### ISSUE TITLE
Need a test for "redis://...

### ISSUE THREAD

**Issue by steveklabnik** [resque/resque#971]
Coveralls shows that we have no test for building something with "redis:// https://coveralls.io/builds/24478/source?filename=lib%2Fresque%2Fconfig.rb

So I tried to write a test:

``` diff
diff --git a/test/resque/config_test.rb b/test/resque/config_test.rb
index 7585826..cd93eab 100644
--- a/test/resque/config_test.rb
+++ b/test/resque/config_test.rb
@@ -12,6 +12,12 @@ describe Resque::Config do
       assert_equal 'namespace', config.redis.namespace
     end

+    it "works correctly with a URI string" do
+      redis = Redis.new(:url => 'redis://127.0.0.1:6379')
+      config.redis = 'redis://127.0.0.1:6379'
+      assert config.redis
+    end
+
     it "works correctly with a Redis::Namespace param" do
       new_redis = Redis.new(:host => "localhost", :port => 9736)
       new_namespace = Redis::Namespace.new("namespace", :redis => new_redis)
@@ -45,4 +51,5 @@ describe Resque::Config do
       assert_equal config.redis_id, "redis://localhost:6379/0, redis://localhost:6380/0"
     end
   end
-end
\ No newline at end of file
+end
+
```

This doesn't go well. If we make [this line](https://github.com/resque/resque/blob/2b01ac6c6ac606a418c7a25bf10aab73cc54c42a/lib/resque/config.rb#L20) return nil, then the test still passes, because of https://github.com/resque/redis-namespace/blob/master/lib/redis/namespace.rb#L192 in redis-namespace 

Tl;DR: I don't know how to write a good test for this.
